internal/input.go

```diff
@@ -0,0 +1,206 @@
+package internal
+
+import (
+	"errors"
+	"fmt"
+	"github.com/influxdata/influxdb/v2/task/options"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/tcnksm/go-input"
+)
+
+// vt100EscapeCodes
+var (
+	KeyEscape   = byte(27)
+	ColorRed    = []byte{KeyEscape, '[', '3', '1', 'm'}
+	ColorYellow = []byte{KeyEscape, '[', '3', '3', 'm'}
+	ColorCyan   = []byte{KeyEscape, '[', '3', '6', 'm'}
+	KeyReset    = []byte{KeyEscape, '[', '0', 'm'}
+)
+
+func PromptWithColor(s string, color []byte) []byte {
+	bb := append(color, []byte(s)...)
+	return append(bb, KeyReset...)
+}
+
+func GetConfirm(ui *input.UI, promptFunc func() string) bool {
+	prompt := PromptWithColor("Confirm? (y/n)", ColorRed)
+	for {
+		ui.Writer.Write(PromptWithColor(promptFunc(), ColorCyan))
+		result, err := ui.Ask(string(prompt), &input.Options{
+			HideOrder: true,
+		})
+		if err != nil {
+			return false
+		}
+		switch result {
+		case "y":
+			return true
+		case "n":
+			return false
+		default:
+			continue
+		}
+	}
+}
+
+var errPasswordNotMatch = fmt.Errorf("passwords do not match")
+
+var errPasswordIsTooShort error = fmt.Errorf("password is too short")
+
+func GetSecret(ui *input.UI) (secret string) {
+	var err error
+	query := string(PromptWithColor("Please type your secret", ColorCyan))
+	for {
+		secret, err = ui.Ask(query, &input.Options{
+			Required:  true,
+			HideOrder: true,
+			Hide:      true,
+			Mask:      false,
+		})
+		switch err {
+		case input.ErrInterrupted:
+			os.Exit(1)
+		default:
+			if secret = strings.TrimSpace(secret); secret == "" {
+				continue
+			}
+		}
+		break
+	}
+	return secret
+}
+
+func GetPassword(ui *input.UI, showNew bool) (password string) {
+	newStr := ""
+	if showNew {
+		newStr = " new"
+	}
+	var err error
+enterPassword:
+	query := string(PromptWithColor("Please type your"+newStr+" password", ColorCyan))
+	for {
+		password, err = ui.Ask(query, &input.Options{
+			Required:  true,
+			HideOrder: true,
+			Hide:      true,
+			Mask:      false,
+			ValidateFunc: func(s string) error {
+				if len(s) < 8 {
+					return errPasswordIsTooShort
+				}
+				return nil
+			},
+		})
+		switch err {
+		case input.ErrInterrupted:
+			os.Exit(1)
+		case errPasswordIsTooShort:
+			ui.Writer.Write(PromptWithColor("Password too short - minimum length is 8 characters!\n\r", ColorRed))
```

Comment:
nit: no need to add exclamation at the end: **!**

Reply:
ok, This came from original cmd/influx/setup.go :)

Reply:
I figured it probably did 👍